### PR TITLE
Update version to 4.0 to prep for release candidate

### PR DESCRIPTION
**Pull request by krlberry**
## Description
Updates version number to 4.0 and release type to "beta" to prep for the 4.0 release candidate. My plan was to pull off a 4.0 branch after this is merged. (And after we can check the Jenkins tests for successes / failures.) 

## Related Issue
Do I need to open a ticket for a release candidate? 

## Motivation and Context
ISIS 4.0 Release Candidate creation. 

## How Has This Been Tested?
No testing.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by scsides — DISMISSED** [review]
That's a good question. In the past we had blanket approval to not need a ticket for changes directly related to the release, aka, these.

**Comment by jessemapel** [comment]
We likely need to update our PR guidelines to account for PRs like this and testing updates that are not related to issues, but are just general maintenance on the project.

**Review by kaitlyndlee — APPROVED** [review]

File: isis/CMakeLists.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 # Specify the required version of CMake.  If your machine does not
 #  have this, it should be easy to build from https://cmake.org/download/
-cmake_minimum_required(VERSION 3.10)
+#cmake_minimum_required(VERSION 3.10)
```
Comment (jessemapel) [inline isis/CMakeLists.txt:8]: Why did this get axed? we need 3.10 for ctest/gtest integration

**Review by scsides — APPROVED** [review]

File: isis/CMakeLists.txt (hunk 1 above)
Comment (krlberry) [reply isis/CMakeLists.txt:8]: Oh no, this was unintentional. Thanks for catching it.

**Pull request merged**